tests/test_helpers.c

```diff
@@ -0,0 +1,157 @@
+
+#include "test_helpers.h"
+
+/* HQC PRNG implementation */
+
+// State for HQC PRNG. Analogue of DRBG_ctx in rand/rand_nist.c
+static OQS_SHA3_shake256_inc_ctx hqc_prng_state = { NULL };
+
+// Allocate the state.
+static void hqc_prng_new(void) {
+	OQS_SHA3_shake256_inc_init(&hqc_prng_state);
+}
+
+// entropy_input must have length 48.
+// If personalization_string is non-null, its length must also be 48.
+static void hqc_prng_seed(const uint8_t *entropy_input, const uint8_t *personalization_string) {
+	uint8_t domain = 1;
+	// reset state
+	OQS_SHA3_shake256_inc_ctx_reset(&hqc_prng_state);
+	OQS_SHA3_shake256_inc_absorb(&hqc_prng_state, entropy_input, 48);
+	if (personalization_string != NULL) {
+		OQS_SHA3_shake256_inc_absorb(&hqc_prng_state, personalization_string, 48);
+	}
+	OQS_SHA3_shake256_inc_absorb(&hqc_prng_state, &domain, 1);
+	OQS_SHA3_shake256_inc_finalize(&hqc_prng_state);
+}
+
+// random_array must have length bytes_to_read.
+static void hqc_prng_randombytes(uint8_t *random_array, size_t bytes_to_read) {
+	OQS_SHA3_shake256_inc_squeeze(random_array, bytes_to_read, &hqc_prng_state);
+}
+
+static void hqc_prng_get_state(void *out) {
+	OQS_SHA3_shake256_inc_ctx_clone((OQS_SHA3_shake256_inc_ctx *)out, &hqc_prng_state);
+}
+
+static void hqc_prng_set_state(const void *in) {
+	OQS_SHA3_shake256_inc_ctx_clone(&hqc_prng_state, (const OQS_SHA3_shake256_inc_ctx *)in);
+}
+
+static void hqc_prng_free(OQS_KAT_PRNG_state *saved_state) {
+	if (saved_state != NULL) {
+		OQS_SHA3_shake256_inc_ctx_release(&saved_state->hqc_state);
+	}
+	if (hqc_prng_state.ctx != NULL) {
+		OQS_SHA3_shake256_inc_ctx_release(&hqc_prng_state);
+		hqc_prng_state.ctx = NULL;
+	}
+}
+
+/* Helpers for identifying algorithms */
+static int is_mceliece(const char *method_name) {
+	return ( !strcmp(method_name, OQS_KEM_alg_classic_mceliece_348864)
+	         || !strcmp(method_name, OQS_KEM_alg_classic_mceliece_348864f)
+	         || !strcmp(method_name, OQS_KEM_alg_classic_mceliece_460896)
+	         || !strcmp(method_name, OQS_KEM_alg_classic_mceliece_460896f)
+	         || !strcmp(method_name, OQS_KEM_alg_classic_mceliece_6688128)
+	         || !strcmp(method_name, OQS_KEM_alg_classic_mceliece_6688128f)
+	         || !strcmp(method_name, OQS_KEM_alg_classic_mceliece_6960119)
+	         || !strcmp(method_name, OQS_KEM_alg_classic_mceliece_6960119f)
+	         || !strcmp(method_name, OQS_KEM_alg_classic_mceliece_8192128)
+	         || !strcmp(method_name, OQS_KEM_alg_classic_mceliece_8192128f) );
+}
+
+static int is_hqc(const char *method_name) {
+	return (0 == strcmp(method_name, OQS_KEM_alg_hqc_128))
+	       || (0 == strcmp(method_name, OQS_KEM_alg_hqc_192))
+	       || (0 == strcmp(method_name, OQS_KEM_alg_hqc_256));
+}
+
+/* OQS_KAT_PRNG interface implementation */
+
+OQS_KAT_PRNG *OQS_KAT_PRNG_new(const char *method_name) {
+	OQS_KAT_PRNG *prng = malloc(sizeof(OQS_KAT_PRNG));
+	if (prng != NULL) {
+		prng->max_kats = is_mceliece(method_name) ? 10 : 100;
```

Comment:
I thought "all" means 100 KATs? Why only 10 for McEliece?

Reply:
The McEliece NIST submission only provides 10 KATs.